### PR TITLE
Feat/add shared state

### PR DESCRIPTION
**Pull request by StuffbyYuki**
Modified the script so that:
- You can now select either use postgres or duckdb for the state db.
- If you specify `GATEWAY` variable in `.env` var or `-e GATEWAY=shared_state` in docker run commands, you'll use a postgres db for the state db. It'll use duckdb otherwise. 
- I have a free tier managed postgres set up, so let me know if you want to use that to give this a try. 

Also, this PR includes a small change [this other PR made](https://github.com/UN-OSAA/osaa-mvp/pull/15).

### CODE REVIEW

**Review by db-tycoon-stephen — COMMENTED** [review]
I haven't been able to run this completely because my containers have been terminating before the job finishes. Not sure if this is just happening with me or if its related to this PR.
image

**Comment by StuffbyYuki** [comment]
@db-tycoon-stephen I think we just need to add postgres configs in docker compose yml. I'm not at my desk but can take a look on Monday

File: entrypoint.sh [hunk 1]
```diff
@@ -7,7 +7,7 @@ case "$1" in
     ;;
   "transform")
     cd sqlMesh
-    sqlmesh plan --auto-apply --include-unmodified --create-from prod --no-prompts "${TARGET:-dev}"
+    sqlmesh --gateway "${GATEWAY}:-local" plan --auto-apply --include-unmodified --create-from prod --no-prompts "${TARGET:-dev}"
```
Comment (db-tycoon-stephen) [inline entrypoint.sh:10]: I was experimenting with llms and it made this suggestion for this line and the others like it. It ended up solving the problem for me.
```suggestion
    sqlmesh --gateway "${GATEWAY:-local}" plan --auto-apply --include-unmodified --create-from prod --no-prompts "${TARGET:-dev}"
```

Comment (StuffbyYuki) [reply entrypoint.sh:10]: @db-tycoon-stephen This is the EXACT change I made that I forgot to push on Friday! Good thing you caught it. The curly braces need to wrap around the whole thing, not only the term 'GATEWAY'

Comment (StuffbyYuki) [reply entrypoint.sh:10]: I don't which one won't make a mess, do I just push my commit or just approve your change.

Comment (db-tycoon-stephen) [reply entrypoint.sh:10]: Push yours since I only made the suggested fix in this one spot. Once it's commited I'll resolve this

Comment (StuffbyYuki) [reply entrypoint.sh]: Just pushed the change. Please review!

**Review by sciortino — APPROVED** [review]
Approved!

**Review by db-tycoon-stephen — APPROVED** [review]

**Review by db-tycoon-stephen — APPROVED** [review]
Approved!

**Pull request merged**